workspaces/ballerina/ballerina-extension/src/utils/auth-utils.ts

```diff
@@ -0,0 +1,155 @@
+ */
+export function isCloudEditorEnvironment(): boolean {
+    return !!process.env.CLOUD_STS_TOKEN;
+}
+
+/**
+ * Reads the current access token from Settings.toml
+ * @returns Promise<string | undefined> - The current access token or undefined if not found
+ */
+export async function getCurrentAccessToken(): Promise<string | undefined> {
+    try {
+        const settingsPath = path.join(os.homedir(), '.ballerina', 'Settings.toml');
+        
+        if (!fs.existsSync(settingsPath)) {
+            return undefined;
+        }
+
+        const content = fs.readFileSync(settingsPath, 'utf8');
+        const tokenMatch = content.match(/accesstoken\s*=\s*"([^"]*)"/);
+        
+        return tokenMatch ? tokenMatch[1] : undefined;
+    } catch (error) {
+        console.error('Failed to read current access token:', error);
+        return undefined;
+    }
+}
+
+/**
+ * Checks if a JWT token is expired or about to expire (within 5 minutes)
+ * @param token - The JWT token to check
+ * @returns boolean - true if token is expired or about to expire
+ */
+export function isTokenExpiredOrExpiring(token: string): boolean {
+    try {
+        // JWT tokens have 3 parts separated by dots
+        const parts = token.split('.');
+        if (parts.length !== 3) {
+            return true; // Not a valid JWT format
+        }
+
+        // Decode the payload (second part)
+        const payload = JSON.parse(atob(parts[1]));
+        
+        if (!payload.exp) {
+            return true; // No expiration time means we should refresh
+        }
+
+        // Check if token expires within 5 minutes (300 seconds)
+        const currentTime = Math.floor(Date.now() / 1000);
+        const expirationTime = payload.exp;
+        const bufferTime = 300; // 5 minutes buffer
+        
+        return (expirationTime - currentTime) <= bufferTime;
+    } catch (error) {
+        console.warn('Failed to decode token, treating as expired:', error);
+        return true; // If we can't decode it, treat as expired
+    }
+}
+
+/**
+ * Checks if the token needs to be updated based on current vs new token
+ * @param currentToken - The token currently in Settings.toml
+ * @param newToken - The new STS token from platform extension
+ * @returns boolean - true if token needs updating
+ */
+export function shouldUpdateToken(currentToken: string | undefined, newToken: string): boolean {
+    // No current token exists
+    if (!currentToken) {
+        return true;
+    }
+
+    // Tokens are different
+    if (currentToken !== newToken) {
+        return true;
+    }
+
+    // Current token is expired or expiring soon
+    if (isTokenExpiredOrExpiring(currentToken)) {
+        return true;
+    }
+
+    return false;
+}
+
+/**
+ * Updates the Ballerina Settings.toml file with the provided STS token
+ * @param stsToken - The STS token to set as accesstoken
+ */
+export async function updateBallerinaSettingsWithStsToken(stsToken: string): Promise<void> {
```

Comment:
we can consider using `@iarna/toml` to parse toml,
as we are already using it in MI and Platform extensions